### PR TITLE
feat: adds watcher script

### PR DESCRIPTION
**Pull request by b0dhidharma**
We still need to:
- Define a mempool service (mainnet, and kovan testing)
- Define a way of reporting hashes (flashbots? or just pay coinbase publicly?)

### CODE REVIEW

File: hardhat.config.ts [hunk 1]
```diff
@@ -16,10 +17,19 @@ module.exports = {
...
+          },
         },
         localMainnet: {
           url: process.env.LOCAL_MAINNET_HTTPS_URL,
-          accounts: [process.env.LOCAL_MAINNET_PRIVATE_KEY],
+          // accounts: [process.env.LOCAL_MAINNET_PRIVATE_KEY],
+        },
+        kovan: {
+          url: process.env.KOVAN_HTTPS_URL,
+          accounts: [process.env.KOVAN_PRIVATE_KEY, '0x8901af9255b653e9a8f654d84b53d37b9134eb5e949f394c8ddb0c2ef4481287'],
```
Comment (lbertenasco) [inline hardhat.config.ts:31]: Remove PK (env variable)

File: scripts/watcher/index.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    validating: false,
+  };
+}
+
+async function reportHash(hash: string, gasPrice: BigNumber): Promise<void> {
+  console.log('reporting hash', hash);
+  await stealthVault.reportHash(hash, { gasPrice: gasPrice.mul(3) });
+  // nonce++;
+  // const populatedTx = await stealthVault.populateTransaction.reportHash(hash, { gasLimit: 1000000, gasPrice: gasPrice.mul(3), nonce })
+  // const wall = new Wallet('0x8901af9255b653e9a8f654d84b53d37b9134eb5e949f394c8ddb0c2ef4481287');
```
Comment (lbertenasco) [inline scripts/watcher/index.ts]: remvoe comment or PK (use env)

File: scripts/watcher/index.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+import { ethers } from 'hardhat';
+import _ from 'lodash';
+import StealthVault from '../../artifacts/contracts/StealthVault.sol/StealthVault.json';
+import { BigNumber, Contract, Transaction as EthersTransaction, utils, Wallet } from 'ethers';
+import { createAlchemyWeb3 } from '@alch/alchemy-web3';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+
+// Using WebSockets
+const web3 = createAlchemyWeb3('wss://eth-kovan.ws.alchemyapi.io/v2/FxGE-9nBVx0yYsI3IqesIcXxgxbxe2bu');
+// const alchemyProvider = new ethers.providers.AlchemyWebSocketProvider('kovan', 'FxGE-9nBVx0yYsI3IqesIcXxgxbxe2bu');
```
Comment (lbertenasco) [inline scripts/watcher/index.ts]: remove alchemy api key (use env and deprecate it from alchemy just in case)

**Review by lbertenasco — CHANGES_REQUESTED** [review]

File: scripts/watcher/simple.ts [hunk 4]
```diff
@@ -0,0 +1,116 @@
+import { Transaction as Web3Transaction } from 'web3-core';
+import { ethers } from 'hardhat';
+import _ from 'lodash';
+import StealthRelayer from '../../artifacts/contracts/StealthRelayer.sol/StealthRelayer.json';
+import { BigNumber, Contract, Transaction as EthersTransaction } from 'ethers';
+import { createAlchemyWeb3 } from '@alch/alchemy-web3';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+
+// Using WebSockets
+const web3 = createAlchemyWeb3('wss://eth-kovan.ws.alchemyapi.io/v2/OAh_8Jbu8aMsuFAj1n8gRzo8PPRfK7VP');
```
Comment (lbertenasco) [inline scripts/watcher/simple.ts:10]: beware of api key

**Comment by b0dhidharma** [comment]
For some reason this code inspector mother fucker keeps checking pull requests, that no one asked him to. @lbertenasco please review it a bit. 
Final watcher would be: `yearn.ts` its ofc going to be a bit changed when going into prod (addresses, and taking out the api key from the file).

File: package.json [hunk 5]
```diff
@@ -39,41 +39,43 @@
...
     "lodash": "4.17.21",
-    "mocha": "8.4.0",
+    "mocha": "9.0.1",
     "moment": "2.29.1",
     "pinst": "2.1.6",
-    "prettier": "2.3.0",
-    "prettier-plugin-solidity": "1.0.0-beta.11",
+    "prettier": "2.3.1",
+    "prettier-plugin-solidity": "1.0.0-beta.13",
     "solc-0.8": "npm:solc@^0.8.4",
```
Comment (lbertenasco) [inline package.json:79]: what is this?

**Review by lbertenasco — APPROVED** [review]

File: package.json (hunk 5 above)
Comment (alejoamiras) [reply package.json:79]: prettier ? or solc-0.8 ? prettier is for styling code, and solc-0.8 is used for creating automatic documentation (not being used right now). But its an official repo afaik.

**Pull request merged**